src/battle-scene.ts

```diff
@@ -2506,32 +2507,46 @@ export default class BattleScene extends SceneBase {
    * @returns
    */
   getMysteryEncounter(override: MysteryEncounter): MysteryEncounter {
+    const biomeMysteryEncounters = mysteryEncountersByBiome.get(this.arena.biomeType);
     // Loading override or session encounter
     let encounter: MysteryEncounter;
-    if (!Utils.isNullOrUndefined(Overrides.MYSTERY_ENCOUNTER_OVERRIDE) && Overrides.MYSTERY_ENCOUNTER_OVERRIDE < allMysteryEncounters.length) {
-      encounter = allMysteryEncounters[Overrides.MYSTERY_ENCOUNTER_OVERRIDE];
+    if (!Utils.isNullOrUndefined(Overrides.MYSTERY_ENCOUNTER_OVERRIDE) && Overrides.MYSTERY_ENCOUNTER_OVERRIDE < biomeMysteryEncounters.length) {
+      encounter = biomeMysteryEncounters[Overrides.MYSTERY_ENCOUNTER_OVERRIDE];
     } else {
-      encounter = override?.encounterType >= 0 ? allMysteryEncounters[override?.encounterType] : null;
+      encounter = override?.encounterType >= 0 ? biomeMysteryEncounters[override?.encounterType] : null;
     }
 
     // Generate new encounter if no overrides
     if (!encounter) {
       const tierValue = Utils.randSeedInt(64);
       let tier = tierValue > 32 ? MysteryEncounterTier.COMMON : tierValue > 16 ? MysteryEncounterTier.UNCOMMON : tierValue > 6 ? MysteryEncounterTier.RARE : MysteryEncounterTier.SUPER_RARE;
       let availableEncounters = [];
-
+      for (encounter of biomeMysteryEncounters) {
+        console.log("-------" + MysteryEncounterType[encounter.encounterType] + " Encounter Check -------");
+        console.log(encounter);
+        console.log( "Encountercheck: " + encounter.meetsRequirements(this));
+        console.log( "protagCheck: " +  encounter.meetsProtagonistRequirementAndProtagonistPokemonSelected(this));
```

Comment:
Instead of having 3 separate function calls for requirements. Why not have the protag and supporting requirements be checked inside meetsRequirements()?

Reply:
Yeah, that's fine as long as we maintain the ordering; supporting Pokemon must be checked before protagonist.

Reply:
I think the logging is useful to sanity check our event triggers occurring as expected right now though